### PR TITLE
add fastqc part

### PR DESCRIPTION
**Pull request by WenjinGudaisy**
add fastqc part before kallisto

### CODE REVIEW

File: Snakefile [hunk 1]
```diff
@@ -32,14 +32,29 @@ rule fastq_dump:
         fastq-dump -O $(dirname {output.R1}) --split-files --gzip {wildcards.SRR_ID}
         """
 
+rule fastqc:
+    input:
+        R1 = int_dir / "fastqs" / "{SRR_ID}" / "{SRR_ID}_1.fastq.gz",
+        R2 = int_dir / "fastqs" / "{SRR_ID}" / "{SRR_ID}_2.fastq.gz",
+    output:
+        fastqc_1 = int_dir / "fastqc" / "{SRR_ID}" / "{SRR_ID}_1.fastq" /,
```
Comment (jeremykap) [inline Snakefile]: My understanding of fastqc is that its output is HTML. Does this not fail when it can't find the output?

File: Snakefile [hunk 2]
```diff
@@ -32,14 +32,29 @@ rule fastq_dump:
...
+        fastqc_2 = int_dir / "fastqc" / "{SRR_ID}" / "{SRR_ID}_2.fastq" /,
+    shell:
+        """
+        mkdir -p $(dirname {fastqc_1})
+        mkdir -p $(dirname {fastqc_2})
```
Comment (jeremykap) [inline Snakefile]: The outputs are in the same directory so you only have to do this once.

File: Snakefile [hunk 3]
```diff
@@ -32,14 +32,29 @@ rule fastq_dump:
...
+        fastqc -o {output.fastqc_1} {input.R1}
+        fastqc -o {output.fastqc_2} {input.R2}
```
Comment (jeremykap) [inline Snakefile]: I think the -o command takes a directory, not a file path so you'll need to do something like $(dirname {output.fastqc_1}) here.

**Review by jeremykap — CHANGES_REQUESTED** [review]
Thanks for putting this together - did you run it? By eye I think it'll run, but the pipeline will say it will fail if it can't find the HTML files. Can chat more about it when we meet.

**Comment by WenjinGudaisy** [comment]
I am running it right now. Let me push you the latest version later.

**Pull request merged**